### PR TITLE
Kbc 2735 work with gcs file storage

### PR DESCRIPTION
**Pull request by fejjo**
Jira: KBC-2788
MainPR: https://github.com/keboola/connection/pull/3854

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [ ] major (BC break)
  - [x] minor (new feature)
  - [ ] patch (backwards compatible fix)
  - [ ] no release (just test changes)

### CODE REVIEW

File: src/Keboola/StorageApi/GCSUploader.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        if (!$file) {
+            throw new ClientException("Cannot open file {$file}");
+        }
+        $retBucket->upload(
+            $file,
+            [
+                'name' => $filePath,
+                'metadata' => [
+                    // in gcs is not possible set life cycles to directory, it must set by storageClass mapped to life cycle when file is uploaded
+                    'storageClass' => $isPermanent ? self::STORAGE_CLASS_PERMANENT : self::STORAGE_CLASS_STANDARD,
```
Comment (pivnicek) [inline src/Keboola/StorageApi/GCSUploader.php:64]: 👍

File: tests/File/GcsFileTest.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    public function uploadData(): array
+    {
+        $path = __DIR__ . '/../_data/files.upload.txt';
+        $largeFilePath = sys_get_temp_dir() . '/large_abs_upload.txt';
+        $this->generateFile($largeFilePath, 16);
+
+        return [
+            [
+                $path,
+                (new FileUploadOptions())->setIsPublic(true),
```
Comment (pivnicek) [inline tests/File/GcsFileTest.php:63]: hmm, does this test anything?

**Review by pivnicek — APPROVED** [review]
parada,  just not sure if some of the test conditions make sense in the GCS case

File: tests/File/GcsFileTest.php (hunk 2 above)
Comment (fejjo) [reply tests/File/GcsFileTest.php:63]: yeah that test, if I set public true file will be still isPublic = false

File: tests/File/GcsFileTest.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            ],
+            [
+                $path,
+                (new FileUploadOptions())
+                    ->setIsPublic(true),
+            ],
+            [
+                $path,
+                (new FileUploadOptions())
+                    ->setIsEncrypted(false),
```
[the remark this answers is not in the document]
Comment (fejjo) [reply tests/File/GcsFileTest.php:73]: and the same situation here, if I set encrypted false that test the isEncrypted will be always true

**Comment by fejjo** [comment]
@pivnicek one more I add downloadFIle https://github.com/keboola/storage-api-php-client/pull/826/commits/b636f0f22d56bda86143f09a06fd31df6b0f432b that should be last update thx

File: tests/File/GcsFileTest.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (pivnicek) [reply tests/File/GcsFileTest.php:73]: ah, I see, good point

**Pull request merged**